src/metrics/postgres_metric.ts

```
@@ -0,0 +1,74 @@
+import { AbstractMetric, Datasource, MetricId, MetricQuery } from './metric';
+
+import * as _ from 'lodash';
+export class PostgresMetric extends AbstractMetric {
+
+  private _targetName; //save first target name, while multi metric not implemented
+
+  constructor(datasource: Datasource, targets: any[], id?: MetricId) {
+    super(datasource, targets, id);
+    this._targetName = targets[0].refId;
```

lets throw an exception if `targets.length === 0`